Source/Plugins/CoherenceViewer/CoherenceNode.h

```diff
@@ -94,7 +94,7 @@ class CoherenceNode : public GenericProcessor, public Thread
     AtomicScopedWritePtr<std::vector<std::vector<double>>> coherenceWriter;
     AtomicScopedReadPtr<Array<FFTWArray>> dataReader;
 
-    CumulativeTFR* TFR;
+    ScopedPointer<CumulativeTFR> TFR;
```

Comment:
This avoids a memory leak when you create a new `CumulativeTFR` in `updateSettings`, since it automatically frees the old pointer when it's assigned a new one.

Ultimately though I think it would make more sense to allow changing settings in `CumulativeTFR` itself - that way we wouldn't need to make a new one every time we update settings and this wouldn't need to be a pointer at all.

Reply:
Cool yes, I forgot we did this before on something! 

Yeah I had that in at one time, but I ended up calling the init function anyways to reset the size of FFTWarrays. I am planning on add a function to change settings for things that don't affect FFTW.